### PR TITLE
Make sure the saved TGraph is a proper C++ variable

### PR DESCRIPTION
**Pull request by couet**
Make  sure the saved TGraph is a proper C++ variable.
As suggested here: https://its.cern.ch/jira/browse/ROOT-8277
it works also for the classes inheriting from TGraph

### CODE REVIEW

**Review by dpiparo — COMMENTED** [review]
This is great, @couet , thanks. Before approving, would it be perhaps good to have a test for this feature?

**Comment by couet** [comment]
Indeed it is just using `gInterpreter->MapCppName` if a test should be done it would be to test `MapCppName`.  I thought a bit about it and I think this issue might be also seen with some other objects like histograms. I plan to check that.

**Comment by dpiparo** [comment]
Is mangling supposed to address those characters too? Maybe there is something simple which could be done for names of objects in ROOT, for example keeping only letters and numbers and postponing the hash of the name? In the end we need this for autogenerated code, but maybe I am wrong.

**Comment by dpiparo** [comment]
Hi @couet , are we still missing any ancillary developments to proceed with this PR?

**Comment by couet** [comment]
I am on it. It is more complex than expected. TH2Poly is more complex

File: hist/hist/src/TGraph.cxx [hunk 1]
```diff
@@ -2189,7 +2189,8 @@ void TGraph::SaveHistogramAndFunctions(std::ostream &out, const char *varname, I
       TString hname = fHistogram->GetName();
       fHistogram->SetName(TString::Format("Graph_%s%d", hname.Data(), frameNumber).Data());
       fHistogram->SavePrimitive(out, "nodraw");
-      out << "   "<<varname<<"->SetHistogram(" << fHistogram->GetName() << ");" << std::endl;
+      out << "   " <<varname << "->SetHistogram(" << gInterpreter->MapCppName(fHistogram->GetName()) << ");"
```
Comment (linev) [inline hist/hist/src/TGraph.cxx:2192]: Not sure if `gInterpreter->MapCppName` required here.

Histogram name is set as specified before - and only can be find via that name.

File: hist/hist/src/TGraph.cxx [hunk 2]
```diff
@@ -2152,7 +2152,7 @@ void TGraph::SavePrimitive(std::ostream &out, Option_t *option /*= ""*/)
 
 TString TGraph::SaveArray(std::ostream &out, const char *suffix, Int_t frameNumber, Double_t *arr)
 {
-   const char *name = GetName();
+   const char *name = gInterpreter->MapCppName(GetName());
    if (!name || !*name)
```
Comment (linev) [inline hist/hist/src/TGraph.cxx:2156]: One need to use `TString name` - if one store output of `MapCppName` method. Like: 
```
TSting name = gInterpreter->MapCppName(GetName());
if (name.IsNull()) name = "Graph";
```

**Review by linev — CHANGES_REQUESTED** [review]

File: hist/hist/src/TGraph.cxx (hunk 1 above)
Comment (linev) [reply hist/hist/src/TGraph.cxx:2192]: I see now code of `TH1::SavePrimitive`. Yes, one should keep here `gInterpreter->MapCppName`

**Review by linev — APPROVED** [review]
Only one place with `TString` need to be adjusted.

File: hist/hist/src/TGraph.cxx (hunk 2 above)
Comment (couet) [reply hist/hist/src/TGraph.cxx:2156]: I'll check. Thanks

**Pull request merged**